osutil/disks/disks.go

```diff
@@ -0,0 +1,282 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// diskFromMountPoint is exposed for mocking from other tests via
+// MockMountPointDisksToPartionMapping, but we can't just assign
+// diskFromMountPointImpl to diskFromMountPoint due to signature differences,
+// the former returns a *disk, the latter returns a Disk, and as such they can't
+// be assigned to each other
+var diskFromMountPoint = func(mountpoint string, opts *Options) (Disk, error) {
+	return diskFromMountPointImpl(mountpoint, opts)
+}
+
+// Options is a set of options used when querying information about
+// partition and disk devices.
+type Options struct {
+	// IsDecryptedDevice indicates that the mountpoint is referring to a
+	// decrypted device.
+	IsDecryptedDevice bool
+}
+
+// Disk is a single physical disk device that contains partitions.
+type Disk interface {
+	// FindMatchingPartitionUUID finds the partition uuid for a partition
+	// matching the specified label on the disk. Note that for non-ascii labels
+	// like "Some label", the label should be encoded using \x<hex> for
+	// potentially non-safe characters like in "Some\x20Label".
```

Comment:
IIRC there is some code to convert label to the safe-escaped string in the gadget package in case that's useful for later PRs.

Reply:
ah you're right I found it now, https://github.com/snapcore/snapd/blob/475acef864523e3a81baf6fac90f8ff2fd3d89e5/gadget/device_linux.go#L146-L164

Thanks, yeah I think we should unify that here, it would be more helpful to just pass a normal Go string and encode it to do proper comparison. I also ended up just using the non encoded version since I didn't want to go through the encoding bits and your comment reminds me I forgot to update the doc-string here too :facepalm: